### PR TITLE
Golang/add data-types content

### PR DESCRIPTION
**Pull request by jochenrui**
### Description

Hi, it's me again. As part of Hacktoberfest I've found a way to contribute to the open source community by writting my learnings into docs and I like it. This time I added a first iteration of a data-types article for Go with information about the different number types, strings and booleans.

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/go/concepts/data-types/data-types.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+- Unsigned (`uint`) integers ranging from 0 to positive infinity.
+
+Each integer value comes in 4 different memory sizes.
+
+- 8-bit: `int8 / uint8`
+- 16-bit: `int16 / uint16`
+- 32-bit: `int32 / uint32`
+- 64-bit: `int64 / uint64`
+- 32- or 64-bit based on system architecture: `int / uint`
+- synonym for int32: `rune`
```
Comment (sonnynomnom) [inline content/go/concepts/data-types/data-types.md:47]: Capitalize first letter:

```suggestion
- Synonym for int32: `rune`
```

File: content/go/concepts/data-types/data-types.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+### Complex Numbers
+
+Complex numbers are composed of a real number and an imaginary number. The real number is either a float. There are two different sizes:
+
+- 32-bit float + imaginary number: `complex64`
+- 64-bit float + imaginary number: `complex128`
+
+```go
+  // initialize with complex() taking 2 arguments: 1) a float and 2) a complex number
```
Comment (sonnynomnom) [inline content/go/concepts/data-types/data-types.md:65]: Capitalize first letter in the comment, take out the numbers, and take out all the unnecessary spaces:

```suggestion
// Initialize with complex() taking 2 arguments: a float and a complex number
```

File: content/go/concepts/data-types/data-types.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  var complex complex64 = complex(1, 2) // equals 1 + 2i
```
Comment (sonnynomnom) [inline content/go/concepts/data-types/data-types.md:66]: Go through this and take out all unnecessary spaces:

```suggestion
var complex complex64 = complex(1, 2) // Equals 1 + 2i
```

**Review by sonnynomnom — CHANGES_REQUESTED** [review]
Take out all unnecessary spaces in code blocks and make sure the first letter in comments all capitalized.

**Comment by Dusch4593** [comment]
Hey @jochenrui , just a heads up! We've switched from CircleCI to GitHub Actions! You'll want to merge the latest changes from `Codecademy/docs:main` into your fork and current/active branches. 

In your fork, after committing any changes you're working on, update your local `main` branch:

```
git checkout main
git pull upstream main
git push
```

Then, `checkout` into your local `golang/add-types` branch and do the following:

```
git checkout golang/add-types
git merge main
git push
```

This will catch your remote `main` and `golang/add-types` branches up with the `main` branch of the core `docs` repository and allow the tests to go forward! 😄

**Comment by jochenrui** [comment]
> Hey @jochenrui , just a heads up! We've switched from CircleCI to GitHub Actions! You'll want to merge the latest changes from `Codecademy/docs:main` into your fork and current/active branches.
> 
> In your fork, after committing any changes you're working on, update your local `main` branch:
> 
> ```
> git checkout main
> git pull upstream main
> git push
> ```
> 
> Then, `checkout` into your local `golang/add-types` branch and do the following:
> 
> ```
> git checkout golang/add-types
> git merge main
> git push
> ```
> 
> This will catch your remote `main` and `golang/add-types` branches up with the `main` branch of the core `docs` repository and allow the tests to go forward! 😄

Oh I see, thanks for the information :)

**Comment by sonnynomnom** [comment]
Once you do that @jochenrui, then I can merge it in!

File: content/go/concepts/data-types/data-types.md (hunk 3 above)
Comment (jochenrui) [reply content/go/concepts/data-types/data-types.md]: done :)

**Comment by sonnynomnom** [comment]
Cool cool! Did you do what Brandon mentioned?

**Comment by jochenrui** [comment]
> Cool cool! Did you do what Brandon mentioned?

not yet, gotta find out how to manage 2 github accounts on one machine

**Comment by sonnynomnom** [comment]
Okay, when you are done with that, we should be good to go

**Comment by Dusch4593** [comment]
Hey @jochenrui, fwiw if you get an issue on `upstream` about that remote repository not existing or some such, try this first and then try the git commands I mentioned earlier:

```
git remote add upstream https://github.com/Codecademy/docs
```

**Comment by jochenrui** [comment]
> Hey @jochenrui, fwiw if you get an issue on `upstream` about that remote repository not existing or some such, try this first:
> 
> ```
> git remote add upstream https://github.com/Codecademy/docs
> ```
> 
> Then, try the git commands I mentioned earlier

do u perhaps know what I can do ?

I tried pushing over http and over ssh. Over http it says permission denied, which is ridiculous because it should ask me for my credentials instead. Over ssh it doesn't recognize my private key even though I added it to my ssh agent and added the public ssh key to github.

**Comment by Dusch4593** [comment]
> do u perhaps know what I can do ?
> 
> I tried pushing over http and over ssh. Over http it says permission denied, which is ridiculous because it should ask me for my credentials instead. Over ssh it doesn't recognize my private key even though I added it to my ssh agent and added the public ssh key to github.

Hmmm I'm curious about two things:

1. Which branch are you pushing from?
2. Can you write out the exact `push` statement your are using (for both http and ssh)?

**Comment by jochenrui** [comment]
> > do u perhaps know what I can do ?
> > I tried pushing over http and over ssh. Over http it says permission denied, which is ridiculous because it should ask me for my credentials instead. Over ssh it doesn't recognize my private key even though I added it to my ssh agent and added the public ssh key to github.
> 
> Hmmm I'm curious about two things:
> 
> 1. Which branch are you pushing from?
> 2. Can you write out the exact `push` statement your are using (for both http and ssh)?

uh I can write u the history of commands I tried trying http

```bash
git clone https://github.com/jochenrui/docs.git
cd docs
git checkout main
git remote add upstream https://github.com/Codecademy/docs
git pull upstream main
git push
```

```bash
remote: Permission to jochenrui/docs.git denied to saf.
fatal: unable to access 'https://github.com/jochenrui/docs.git/': The requested URL returned error: 403
```

as for ssh
I basically did the same but used git clone git@github.com:jochenrui/docs.git instead and setup my ssh agent

**Comment by Dusch4593** [comment]
> remote: Permission to jochenrui/docs.git denied to saf.

Who is "saf"?

**Comment by jochenrui** [comment]
> > remote: Permission to jochenrui/docs.git denied to saf.
> 
> Who is "saf"?
> 
> 

I share my computer.. thats why I said I'm having troubles getting 2 github accounts to work at the same time 🙁 I tried adding my Username to the project .git config but it didnt use it

**Comment by Dusch4593** [comment]
> I tried adding my Username to the project .git config but it didnt use it

Hmmm, I wonder. What do you get when you run `git config --list`?

**Comment by sonnynomnom** [comment]
I think I'm just going to merge this one and then take a look afterward.

**Comment by Dusch4593** [comment]
Sounds good. 👍🏻

**Comment by sonnynomnom** [comment]
Congrats @jochenrui it's live! https://www.codecademy.com/resources/docs/go/data-types##Booleans

Made some edits: https://github.com/Codecademy/docs/commit/c117021b50439476cc5f67510752be5ff9a85efe

**Comment by jochenrui** [comment]
So cool, thanks 😁

**Comment by sonnynomnom** [comment]
Hi @jochenrui one last thing, can you update your profile picture on Codecademy?

Screen Shot 2021-11-22 at 2 50 14 PM

We just made a change that we are displaying Codecademy profile instead of GitHub.

Sorry for asking, only asking because you are on the home page!

**Comment by jochenrui** [comment]
@sonnynomnom thanks for the info, will do tomorrow :)

**Pull request merged**